### PR TITLE
qa: enhance labeled perf counters test for cephfs-mirror

### PR DESCRIPTION
**Pull request by joscollin**
Fixes: https://tracker.ceph.com/issues/64486

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by joscollin** [comment]
jenkins test make check

**Comment by joscollin** [comment]
Tests passed: https://pulpito.ceph.com/jcollin-2024-03-06_01:11:45-fs:mirror-main-distro-default-smithi/

**Comment by vshankar** [comment]
@joscollin Could you elaborate a bit in the commit message as to what all checks have been added.

File: qa/tasks/cephfs/test_mirroring.py [hunk 1]
```diff
@@ -626,6 +652,12 @@ def test_cephfs_mirror_cancel_sync(self):
 
         snap_list = self.mount_b.ls(path='d0/.snap')
         self.assertTrue('snap0' not in snap_list)
+
+        # check sync_failures
+        res = self.mirror_daemon_command(f'counter dump for fs: {self.primary_fs_name}', 'counter', 'dump')
+        vafter = res[TestMirroring.PERF_COUNTER_KEY_NAME_CEPHFS_MIRROR_PEER][0]
+        self.assertGreater(vafter["counters"]["sync_failures"], vbefore["counters"]["sync_failures"])
```
Comment (vshankar) [inline qa/tasks/cephfs/test_mirroring.py:659]: Wouldn't it suffice to check if the failure count has increases than validating it being greater than it was before?

Comment (joscollin) [reply qa/tasks/cephfs/test_mirroring.py:659]: Do you mean change it to

`self.assertEquals(vafter["counters"]["sync_failures"], before["counters"]["sync_failures"] + 1)` ?

Comment (vshankar) [reply qa/tasks/cephfs/test_mirroring.py:659]: If this is the first and only failure then > 0 would suffice IMO.

Comment (joscollin) [reply qa/tasks/cephfs/test_mirroring.py]: Updated

**Comment by vshankar** [comment]
@joscollin I also wanted to check if you plan to include similar checks for per-client perf metrics in this PR (or do you plan to send a separate change for that).

**Comment by joscollin** [comment]
> @joscollin I also wanted to check if you plan to include similar checks for per-client perf metrics in this PR (or do you plan to send a separate change for that).

a separate PR would be good right?

**Comment by vshankar** [comment]
> > @joscollin I also wanted to check if you plan to include similar checks for per-client perf metrics in this PR (or do you plan to send a separate change for that).
> 
> a separate PR would be good right?

works for me.

**Comment by joscollin** [comment]
jenkins test make check

**Comment by joscollin** [comment]
jenkins test make check

**Comment by vshankar** [comment]
https://pulpito.ceph.com/?branch=wip-vshankar-testing-20240307.013758

(test run review pending)

**Comment by joscollin** [comment]
rebased

**Comment by joscollin** [comment]
jenkins test make check

**Comment by vshankar** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Main#14th-March-2024

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#14th-March-2024

**Pull request merged**